Classes/Controller/Application/PersonalDataController.php

```diff
@@ -61,8 +51,22 @@ public function editPersonalDataAction(ApplicationB $application)
      */
     public function updatePersonalDataAction(ApplicationB $application)
     {
-
         $this->repository->addOrUpdate($application);
         $this->forward("editQualifications", "Application\\Qualifications", null, ['application' => $application->getUid()]);
     }
+
+    /**
+     * Legacy country repository switch for TYPO3 7
+     *
+     * @return CountryRepository|LegacyCountryRepository
+     * @todo remove this in V2
+     */
+    protected function getCountryRepository()
+    {
+        if (VersionNumberUtility::convertVersionNumberToInteger(TYPO3_version) < 8007000) {
```

Comment:
This could be done in `ext_localconf.php` via `Container::registerImplementation()` instead. Then the application code does not need to know anything about this.

Reply:
Done, could you check if I did this correctly? I decided to not use an interface as this solution is  very intermediate until 2.0 is out.